adagrams/game.py

```diff
@@ -1,11 +1,96 @@
+import random
+import copy
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+SCORE_CHART = {
+        1: ["A", "E", "I", "O", "U", "N", "R", "S", "T"],
+        2: ["D", "G"],
+        3: ["B", "C", "M", "P"],
+        4: ["F", "H", "V", "W", "Y"],
+        5: ["K"],
+        8: ["J", "X"],
+        10: ["Q", "Z"]
+    }
+
+
+
 def draw_letters():
-    pass
+    letters_for_user = []
+    fresh_copy = copy.copy(LETTER_POOL)
+    while len(letters_for_user) < 10:
+        letter = random.choice(list(fresh_copy.keys()))
+        value = fresh_copy[letter]
+        if value > 0:
+            letters_for_user.append(letter)
+            fresh_copy[letter] -= 1
+    return letters_for_user
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    letter_bank_copy = copy.copy(letter_bank)
+    word = word.upper()
+    for letter in word:
+        if letter in letter_bank_copy:
+            letter_bank_copy.remove(letter)
```

Comment:
Let's have a closer look at the time complexity involved with lines 60-62. 

Line 60: a for loop that loops as many times as there are letters in word - O(n)
Line 61: `in` operator that in the worst case scenario has to check every letter in `letter_bank_copy` to see if `letter` is in it - O(n)
Line 62: `remove ` method that in the worst case scenario has to check every letter in `letter_bank_copy` to remove `letter` - O(n)

Rather than using a list of the letters in the hand, could we build a helper data structure (like a dictionary) that could let us look up if a letter was part of a user's hand?

If the keys were the letters and the values were the number of letters in the hand, then you could leverage the fact that key look up in dictionaries is a constant O(1) operation. Additionally, instead of using the remove method, you could decrement the value in the dictionary to keep track of how many letters have been played.